src/hooks/useProfileModalControls.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from "react";
+
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+import { userProfileSelector } from "utils/selectors";
+
+import { setUserProfileData } from "store/actions/UserProfile";
+
+import { useSelector } from "./useSelector";
+import { useDispatch } from "./useDispatch";
+
+export const useProfileModalControls = () => {
+  const dispatch = useDispatch();
+  const selectedUserProfile = useSelector(userProfileSelector);
+
+  const openUserProfileModal = useCallback(
+    (userProfile?: WithId<User>) => {
+      dispatch(setUserProfileData(userProfile));
+    },
+    [dispatch]
+  );
```

Comment:
[NB] If we make this a required prop, what are the 'flow on effects' of that at the 'call sites' where it is used? I expect this is often potentially `undefined` where it is used, which is why we probably implemented it like this here.

I feel like the UX would be strange if for some reason it was undefined, and we clicked on something expecting the profile to open, particularly if the modal was already open, as this would sort of result in it being closed (by setting it to `undefined`). This is likely a rare edgecase though.. so may not really matter in reality.

If there aren't many 'flow on effects' from making this non-optional, then it would be good to change it:

```suggestion
  const openUserProfileModal = useCallback(
    (userProfile: WithId<User>) => {
      dispatch(setUserProfileData(userProfile));
    },
    [dispatch]
  );
```

If there would be a lot of flow on effects, then perhaps we can make the code here more obvious that it's not intended to be able to unset the profile, with something like:

```suggestion
  const openUserProfileModal = useCallback(
    (userProfile?: WithId<User>) => {
      // We can only open the modal when we actually have a userProfile
      if (!userProfile) return
      
      dispatch(setUserProfileData(userProfile));
    },
    [dispatch]
  );
```